x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -714,9 +715,14 @@ static void writeEval(PlanStreamOutput out, Eval eval) throws IOException {
     }
 
     static Enrich readEnrich(PlanStreamInput in) throws IOException {
+        Enrich.Mode m = null;
```

Comment:
Is mode nullable? If so, then we need to handle null when writing/reading it from the wire. I think it should not be null, default to ANY for old clusters.

Reply:
Mode gets initialized to ANY if passed as null, inside the Enrich constructor.